kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -265,6 +266,7 @@ protected TransactionImpl buildTransactionInternal(
           latestSnapshot.get().getMetadata(), // reuse latest metadata
           setTxnOpt,
           Optional.empty(), /* clustering cols=empty */
```

Comment:
If I am a clustering writing without Txn.withClusteringColumn, will it go to this branch? The intent of initialClusteringColumns.isPresent() seems to be changing clustering columns.

Reply:
I created this issue to track it: https://github.com/delta-io/delta/issues/4713